.github/workflows/ci.yaml

```diff
@@ -56,3 +57,25 @@ jobs:
 
       - name: Build tests
         run: cmake --build ${{ env.build_dir }}
+
+  test-gsl:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-14
+    env:
+      build_dir: ${{ github.workspace }}/build
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: brew install fmt postgresql cpp-gsl
```

Comment:
Since it's named `test-gsl` and uses `matrix` strategy, let's make this installation step conditional on the OS and add Ubuntu runner